### PR TITLE
Bump to go 1.20

### PR DESCRIPTION
**Pull request by njhale**
Note: I'm _also_ using this to check some of the GitHub workflows

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+FROM ghcr.io/acorn-io/images-mirror/golang:1.20-alpine AS build
 COPY / /src
 WORKDIR /src
 ARG TAG="v0.0.0-dev"
 ENV CGO_ENABLED=0
 RUN --mount=type=cache,target=/go/pkg --mount=type=cache,target=/root/.cache/go-build \
     go build -o bin/acorn-dns -ldflags "-s -w -X 'github.com/acorn-io/acorn-dns/pkg/version.Tag=${TAG}'" .
 
-FROM alpine:3.16.2 AS base
+FROM ghcr.io/acorn-io/images-mirror/alpine:latest AS base
```
Comment (njhale) [inline Dockerfile:9]: I think we'll want version tags published for this image too @yashgiri

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
           fetch-depth: 0
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.19
+          go-version: '1.20'
```
Comment (njhale) [inline .github/workflows/test.yaml:17]: might need to update https://github.com/acorn-io/acorn-dns/blob/main/.github/renovate.json for this too

File: Dockerfile (hunk 1 above)
Comment (yashgiri) [reply Dockerfile:9]: Hi @cjellick just wanted to confirm with you if it is fine to pick up alpine packages from this ghcr path.

Comment (yashgiri) [reply Dockerfile]: Meanwhile if needed this image can be used
`ghcr.io/acorn-io/images-mirror/alpine:3.16.2`

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**